visual-hexdiff: init at 0.0.53

summersamara opened the pull request:
## Description of changes

Init of hexdiff: Visual hexadecimal difference editor

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

summersamara committed:
visual-hexdiff: init at 0.0.53

erictapen commented on pkgs/by-name/vi/visual-hexdiff/package.nix:
```diff
@@ -0,0 +1,49 @@
+    url = "mirror://ubuntu/pool/universe/h/hexdiff/hexdiff_0.0.53.orig.tar.gz";
+    sha256 = "sha256-M1bmkW63pHlfl9zNWEq0EGN1rpVGo+BTUKM9ot4HWqo=";
+  };
+
```
```suggestion
# Fix compiler error that wants a string literal as format string for `wprintw`
```

erictapen commented on pkgs/by-name/vi/visual-hexdiff/package.nix:
```diff
@@ -0,0 +1,49 @@
+  '';
+
+  patches = [
+    (fetchpatch {
```
```suggestion
    # Some changes the debian/ubuntu developers made over the original source code
    (fetchpatch {
```

erictapen reviewed (commented):
Is there any way to signal the name this package has on Repology? Because currently [they list `hexdiff` as the application introduced here](https://repology.org/project/hexdiff/versions), while Nixpkgs is the only distribution exposing a different hexdiff.

summersamara committed:
Fixup

Co-authored-by: Kerstin <kerstin@erictapen.name>

summersamara committed:
Fixup

Co-authored-by: Kerstin <kerstin@erictapen.name>

FliegendeWurst commented:
> Is there any way to signal the name this package has on Repology?

Yes, by making the necessary changes in https://github.com/repology/repology-rules/. Though in this case we may also need to rename hexdiff in nixpkgs.

SuperSandro2000 commented on pkgs/by-name/vi/visual-hexdiff/package.nix:
```diff
@@ -0,0 +1,50 @@
+    url = "mirror://ubuntu/pool/universe/h/hexdiff/hexdiff_0.0.53.orig.tar.gz";
+    sha256 = "sha256-M1bmkW63pHlfl9zNWEq0EGN1rpVGo+BTUKM9ot4HWqo=";
+  };
+# Fix compiler error that wants a string literal as format string for `wprintw`
+  prePatch = ''
+    substituteInPlace sel_file.c \
+      --replace 'wprintw(win, txt_aide_fs[foo]);' 'wprintw(win, "%s", txt_aide_fs[foo]);'
+  '';
```
```suggestion
  postPatch = ''
  	# Fix compiler error that wants a string literal as format string for `wprintw`
    substituteInPlace sel_file.c \
      --replace-fail 'wprintw(win, txt_aide_fs[foo]);' 'wprintw(win, "%s", txt_aide_fs[foo]);'
  '';
```

SuperSandro2000 commented on pkgs/by-name/vi/visual-hexdiff/package.nix:
```diff
@@ -0,0 +1,50 @@
+  version = "0.0.53";
+
+  src = fetchurl {
+    url = "mirror://ubuntu/pool/universe/h/hexdiff/hexdiff_0.0.53.orig.tar.gz";
```
```suggestion
    url = "mirror://ubuntu/pool/universe/h/hexdiff/hexdiff_${version}.orig.tar.gz";
```

SuperSandro2000 commented on pkgs/by-name/vi/visual-hexdiff/package.nix:
```diff
@@ -0,0 +1,50 @@
+
+  src = fetchurl {
+    url = "mirror://ubuntu/pool/universe/h/hexdiff/hexdiff_0.0.53.orig.tar.gz";
+    sha256 = "sha256-M1bmkW63pHlfl9zNWEq0EGN1rpVGo+BTUKM9ot4HWqo=";
```
```suggestion
    hash = "sha256-M1bmkW63pHlfl9zNWEq0EGN1rpVGo+BTUKM9ot4HWqo=";
```

SuperSandro2000 commented on pkgs/by-name/vi/visual-hexdiff/package.nix:
```diff
@@ -0,0 +1,50 @@
+    maintainers = with maintainers; [ erictapen ];
+    mainProgram = "hexdiff";
+    platforms = platforms.all;
+  };
+
+}
```
```suggestion
  };
}
```

SuperSandro2000 commented on pkgs/by-name/vi/visual-hexdiff/package.nix:
```diff
@@ -0,0 +1,50 @@
+
+  meta = with lib; {
+    description = "Visual hexadecimal difference editor";
+    homepage = "http://tboudet.free.fr/hexdiff/";
```
```suggestion
    homepage = "http://tboudet.free.fr/hexdiff/";
```

SuperSandro2000 commented on pkgs/by-name/vi/visual-hexdiff/package.nix:
```diff
@@ -0,0 +1,50 @@
+  '';
+
+  patches = [
+    # Some changes the debian/ubuntu developers made over the original source code
```
That doesn't explain anything about what the changesare

summersamara replied:
added a link to the associated changelog

summersamara committed:
Use postPatch instead of prePatch to apply fix

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

summersamara committed:
use attribute instead of hardcoded version in package url

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

summersamara committed:
use hash instead of sha256 attribute

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

summersamara committed:
remove empty line

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

summersamara committed:
Add link to the changelogs associated with patch

summersamara committed:
Hardcode version in package url as version attribute doesn't work

reverts https://github.com/NixOS/nixpkgs/pull/327172/commits/dd5fda8f0b8c3fae4bd3cf7898759e377f8f0fe5

summersamara committed:
Fix indent style issue: use spaces not tabs

FliegendeWurst commented on pkgs/by-name/vi/visual-hexdiff/package.nix:
```diff
@@ -0,0 +1,50 @@
+    license = licenses.wtfpl;
+    maintainers = with maintainers; [ erictapen ];
+    mainProgram = "hexdiff";
+    platforms = platforms.all;
```
Please fix or mark as broken for darwin
```
clang -o sel_file.o -c sel_file.c -g -DVERSION=\"0.0.53\" -DTRACE=0  -ansi
sel_file.c:22:7: error: conflicting types for 'strdup'
char *strdup(char *);
      ^
/nix/store/9l0nxr441n8qr0dk5n48m38dz4dswmg5-apple-sdk-10.12.2/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/string.h:117:7: note: previous declaration is here
char    *strdup(const char *__s1);
         ^
1 error generated.
```

summersamara replied:
@FliegendeWurst I pushed a change that might fix the issue

summersamara committed:
Try to fix issue on Darwin

summersamara committed:
fix format

FliegendeWurst commented on pkgs/by-name/vi/visual-hexdiff/package.nix:
```diff
@@ -0,0 +1,57 @@
+    license = licenses.wtfpl;
+    maintainers = with maintainers; [ erictapen ];
+    mainProgram = "hexdiff";
+    platforms = platforms.all;
```
```suggestion
    platforms = platforms.unix;
```
Does not cross-compile for Windows :)

summersamara committed:
update supported platforms

Co-authored-by: Arne Keller <2012gdwu+github@posteo.de>
